crates/compiler/fmt/src/spaces.rs

```diff
@@ -718,6 +729,20 @@ impl<'a> RemoveSpaces<'a> for Expr<'a> {
     }
 }
 
+fn remove_spaces_bad_ident(ident: BadIdent) -> BadIdent {
+    match ident {
+        BadIdent::Start(_) => BadIdent::Start(Position::zero()),
+        BadIdent::Space(e, _) => BadIdent::Space(e, Position::zero()),
+        BadIdent::Underscore(_) => BadIdent::Underscore(Position::zero()),
+        BadIdent::QualifiedTag(_) => BadIdent::QualifiedTag(Position::zero()),
+        BadIdent::WeirdAccessor(_) => BadIdent::WeirdAccessor(Position::zero()),
+        BadIdent::WeirdDotAccess(_) => BadIdent::WeirdDotAccess(Position::zero()),
+        BadIdent::WeirdDotQualified(_) => BadIdent::WeirdDotQualified(Position::zero()),
+        BadIdent::StrayDot(_) => BadIdent::StrayDot(Position::zero()),
+        BadIdent::BadOpaqueRef(_) => BadIdent::BadOpaqueRef(Position::zero()),
+    }
+}
```

Comment:
Do you think there's some system we could employ to easily check for AST equality modulo spacing without having to do this kind of thing for every AST type? I feel like there has to be a better way.. though off the top of my head, nothing better than either checking the canonical AST or something ridiculous like HKTs/pattern calculi. Just bringing up as something we might want to think about.

Reply:
Glad you asked!

The "stack overflow in layout" thing we've been debugging on Zulip, is something I encountered while I've been experimenting with converting the AST + parser (and maybe eventually the formatter) to be code-gen'd from a Roc script.

With this approach it'll be straight-forward to also code-gen things like RemoveSpaces impl.